src/rfactor/process.py

```diff
@@ -226,8 +226,22 @@ def load_rain_file_flanders(file_path, interpolate=False):
     df.loc[df["rain_mm"].isin(nan), "rain_mm"] = np.nan
     df.loc[df["rain_mm"] < 0, "rain_mm"] = np.nan
 
+    if limit is not None:
+        df.loc[df["rain_mm"] > limit, "rain_mm"] = np.nan
+
     if interpolate:
```

Comment:
You will have to add the explaination of the interpolation to the docsting? What is the aim, what is the workflow. Now, it is difficult to entangle what exactly is doen.

What I assume is that you want to fill the array with 0's, once it has more than 36 consecutive NaN's? 

In addition can you check if you can simplify L236-242 with an apply function? You could

```python
df = pd.DataFrame(data = [0,np.nan,np.nan,np.nan,np.nan,0,0,0,np.nan,np.nan,0,0], columns=['value'])
f["condition"] = df["value"].isnull()
df["id"] = df["condition"].diff().cumsum()
df["len"] = 1
df_sum = df.groupby(["id","condition"]).aggregate({"len":"sum"})
df_sum
```

df_sum then returns an id (that is also in df) with the number of consecutive same values. You can use this to filter on true and number of consecutive the same value (in your case 36). Seems easier than the for-loop?

Reply:
The for-loop is 14 lines, the pandas dataframe implementation 7 (with the merge) https://github.com/watem-sedem/rfactor/blob/1b7f71ea903f2611eb3bf3791866b2765a1eaf09/src/rfactor/process.py#L243-L257. We have chosen at the start of the package development to use pandas instead of numpy / for-loops. The diff statement can be shifted to solve the first entry issue.